# **Exploratory Data Analysis**

In [3]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
titanic_df = pd.read_csv("/content/drive/MyDrive/dsml/new_titanic.csv")

**Top 5 datas**

In [7]:
titanic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


**Last 5 datas**

In [8]:
titanic_df.tail()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.00,NaN,S
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.00,B42,S
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,NaN,1,2,W./C. 6607,23.45,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.00,C148,C
890,891,0,3,"Dooley, Mr. Patrick",male,32.0,0,0,370376,7.75,NaN,Q


**Informations of data**

In [9]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


**Statistic Description of data**

In [10]:
titanic_df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


**Count of male and female**

In [11]:
titanic_df.value_counts("Sex")

,count
Sex,
male,577
female,314


**Data types**

In [13]:
titanic_df.dtypes

,0
PassengerId,int64
Survived,int64
Pclass,int64
Name,object
Sex,object
Age,float64
SibSp,int64
Parch,int64
Ticket,object
Fare,float64


**Normalization**

In [16]:
titanic_df.value_counts(["Sex","Survived"])

,,count
Sex,Survived,
male,0,468
female,1,233
male,1,109
female,0,81


here each value is divided by 891

In [18]:
titanic_df.value_counts(["Sex","Survived"],normalize = True,ascending = True)

,,proportion
Sex,Survived,
female,0,0.090909
male,1,0.122334
female,1,0.261504
male,0,0.525253


**Gender  = Female and normalization of their survived value**

In [20]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived",normalize=True,ascending=True)

,proportion
Survived,
0,0.257962
1,0.742038


**unique elements**

In [21]:
titanic_df["Embarked"].nunique()

3

**Using groupby and mean of numeric datas**

In [25]:
titanic_df.groupby(["Embarked","Survived"]).mean(numeric_only = True)

PassengerId    Pclass        Age     SibSp     Parch  \
Embarked Survived                                                         
C        0          409.853333  2.200000  33.666667  0.253333  0.253333   
         1          473.989247  1.634409  28.973671  0.494624  0.451613   
Q        0          480.404255  2.936170  30.325000  0.510638  0.276596   
         1          319.966667  2.866667  22.500000  0.300000  0.000000   
S        0          449.868852  2.545667  30.203966  0.611241  0.348946   
         1          448.857143  1.967742  28.113184  0.493088  0.539171   

                        Fare  
Embarked Survived             
C        0         35.443335  
         1         79.720926  
Q        0         13.335904  
         1         13.182227  
S        0         20.743987  
         1         39.547081

**Covariance**

In [26]:
titanic_df[["Age","Fare"]].cov()

,Age,Fare
Age,211.019125,73.849030
Fare,73.849030,2469.436846


**Correlation**

In [27]:
titanic_df[["Age","Fare"]].corr()

,Age,Fare
Age,1.000000,0.096067
Fare,0.096067,1.000000


**Sample**

In [28]:
sample = titanic_df.sample(n = 100, random_state=42)

In [29]:
sample[["Age","Fare"]].corr()

,Age,Fare
Age,1.000000,0.148739
Fare,0.148739,1.000000


In [30]:
sample[["Age","Fare"]].cov()

,Age,Fare
Age,204.055172,97.056489
Fare,97.056489,1816.599110


In [33]:
sample.shape

(100, 12)

# **Practice Question**

- **Analyze survival by :**
  - Gender
  - Passenger class
  - Age group
  - Fare
- **Based on you findings, describe the characteristics of a passenger with the highest chance of survival.**

In [34]:
titanic_df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

**Survival by Gender**

In [54]:
titanic_df[["Survived","Sex"]].value_counts(normalize=True,ascending=True)

Survived  Sex   
0         female    0.090909
1         male      0.122334
          female    0.261504
0         male      0.525253
Name: proportion, dtype: float64

**So, from the above proportion we can see that the female are likely more to survive than men.**

**Survival by Passenger Class**

In [58]:
titanic_df[["Survived","Pclass"]].value_counts(normalize = True,ascending=True)

Survived  Pclass
0         1         0.089787
1         2         0.097643
0         2         0.108866
1         3         0.133558
          1         0.152637
0         3         0.417508
Name: proportion, dtype: float64

**So, from the above proportion we can see 1st class people are more survived than other classes**

**Survival by Age Group**

In [60]:
titanic_df["AgeGroup"] = pd.cut(
    titanic_df["Age"],
    bins=[0, 12, 18, 40, 60, 80],
    labels=["Child","Teen","Adult","Middle-aged","Senior"]
)

titanic_df[["Survived","AgeGroup"]].value_counts(normalize=True, ascending=True)


Survived  AgeGroup   
1         Senior         0.007003
0         Senior         0.023810
          Child          0.040616
1         Teen           0.042017
          Child          0.056022
0         Teen           0.056022
1         Middle-aged    0.070028
0         Middle-aged    0.109244
1         Adult          0.231092
0         Adult          0.364146
Name: proportion, dtype: float64

**So, from the above proportion or description we can say that Adult group are more likely to survive and high rate of no survival.**

In [63]:
titanic_df["FareGroup"] = pd.cut(
    titanic_df["Fare"],
    bins=[0, 7.91, 14.45, 31, 513],
    labels=["Low", "Medium", "High", "Very High"]
)

titanic_df[["Survived","FareGroup"]].value_counts(normalize=True, ascending=True)


Survived  FareGroup
1         Low          0.049087
          Medium       0.076484
0         Very High    0.106164
1         High         0.116438
0         High         0.144977
1         Very High    0.147260
0         Medium       0.171233
          Low          0.188356
Name: proportion, dtype: float64

**From above description we can asy that high paying fare group people have high chances of surviving**

**Task2: Embarkation Port Analysis**

**The company wants to know which port contributed the most passengers.**

**Tasks:**
- Count passengers from each embarkation port (Embarked)
- Calculate the survival rate for each port
- Identify the port with the highest survival rate

**Counting passengers from each embarked port**

In [68]:
titanic_df.value_counts("Embarked")

,count
Embarked,
S,644
C,168
Q,77


**Survival rate for each port**

In [69]:
titanic_df.groupby("Embarked")["Survived"].mean()

,Survived
Embarked,
C,0.553571
Q,0.389610
S,0.336957


**Port with highest survival rate**

In [70]:
titanic_df.groupby("Embarked")["Survived"].mean().max()

0.5535714285714286

**People from C port has highest survival rate which is 0.5536%**